src/common/util/userTimeout.spec.ts

```diff
@@ -0,0 +1,250 @@
+import {
+    anyRolesMatch,
+    DEFAULT_SESSION_TIMEOUT,
+    getUserSessionTimeout,
+    isRoleMatch,
+    RoleGroupSessionTimeout,
+    sortUserRoles,
+} from './userTimeout'
+
+describe('userTimeout', () => {
+    /**
+     * Note that I deceided to use a Regular Expression matcher here so that we can set the default session timeout via configuration ie. '.',
+     * as we require a different default timeout per application, and require this.toBe( easily configurable.
+     */
+    describe('isRoleMatch()', () => {
```

Comment:
To me this feels like duplication, the underlying functions are already tested as such. See [ArrayPatternMatch](https://github.com/hmcts/rpx-xui-node-lib/blob/master/src/common/util/arrayPatternMatch.spec.ts)

Reply:
Agree. Will remove these.

Reply:
Lol sorry but I disagree with that, they are practically the same tests (though do agree probably not as descriptive but definitely readable)